stages/org.osbuild.tar.meta.json

```diff
@@ -98,7 +98,14 @@
           "description": "Make archive files sparse",
           "type": "boolean",
           "default": false
-        }
+        },
+        "transforms": {
+          "type": "array",
+          "items": {
+            "type": "string",
+	    "description": "Transformation of names during tar(1)"
```

Comment:
Maybe mention that this is passed to `--transform`?

Reply:
Curious if we want to validate this in any way, or restrict the format?

Reply:
I tweaked the description and also fixed the code and tests because when I extended the test I noticed only a single option is allowed. If we want to restrict/validate this we probably need a different approach as tar just calls sed scripts.

We could change it to `rename` and then take a list of path-pairs as input that we can validate. then we would construct the sed expression ourselfs. no strong opinion either way.

Reply:
Torn.  Validating inputs is nice, but so is passing arg values straight to the tool and letting it figure it out.  I think this is one of those situations where the option is [so broad and flexible](https://git.savannah.gnu.org/cgit/tar.git/tree/src/transform.c#n181), there's no way we can validate it properly and I wouldn't want to restrict the option in osbuild to just one use case.  I think I'd rather keep it like it is now (completely open) and leave it to manifest generators to write a proper expression.

Reply:
Yeah, I think if we keep `transform` we can just validate it's a string, anything beyond feels hard. We could have to artificially restrict to e.g. simple renames (and rename the option) to be able to validate I think. I guess it would help people not familiar with how sed works too. But I'm also not sure how much that would help with the validation, we could check that it looks like a valid path but in advance we don't know if the file is there or not.